### PR TITLE
[ci:component:github.com/gardener/machine-controller-manager:v0.52.0->v0.53.0]

### LINKED ISSUE #145: Adoption of Driver.InitializeMachine
**What would you like to be added**:
Implementation Changes needed for Driver.InitializeMachine.
See: 
- https://github.com/gardener/machine-controller-manager/issues/871
- https://github.com/gardener/machine-controller-manager/blob/master/docs/proposals/initialize-machine.md

**Why is this needed**:
`Driver.InitializeMachine` operation has been. AWS provider should do post-creation VM instance initialization activities in this method implementation and not in the VM status check.

### PR DESCRIPTION
**Pull request by gardener-robot-ci-2**
Which issue(s) this PR fixes:
Fixes #145 

**Release Notes**:
```other operator github.com/gardener/machine-controller-manager #907 @rishabh-11
Updated k8s dependencies to `v0.29.3`
```
```other operator github.com/gardener/machine-controller-manager #903 @rishabh-11
Removed vendor directory
```
```breaking developer github.com/gardener/machine-controller-manager #898 @elankath
 New provider method Driver.InitializeMachine added for Post-Creation VM Instance Initialization steps.
```

```feature operator
Add `Ipv6AddressCount` and `Ipv6PrefixCount` to enable the assignment of an ipv6 address and an ipv6 prefix to instances.
```

### CODE REVIEW

**Comment by elankath** [comment]
This is WIP and in testing at the moment and will also incorporate https://github.com/gardener/machine-controller-manager-provider-aws/pull/128

**Comment by axel7born** [comment]
/lgtm

I checked that `Ipv6AddressCount` and `Ipv6PrefixCount` were added and an IPv6 address and an IPv6 prefix were assigned to the instance. To configure the machine class I have used https://github.com/gardener/gardener-extension-provider-aws/pull/785 of the aws provider and the shoot has been setup with `infrastructureConfig.dualStack.enabled=true`.

**Review by rishabh-11 — APPROVED** [review]
/lgtm

File: pkg/aws/core.go [hunk 1]
```diff
@@ -243,17 +247,59 @@ func (d *Driver) CreateMachine(ctx context.Context, req *driver.CreateMachineReq
...
+	targetInstance := instances[0]
+	providerID := encodeInstanceID(providerSpec.Region, *targetInstance.InstanceId)
+
+	svc, err := d.createSVC(request.Secret, providerSpec.Region)
+	if err != nil {
+		return nil, status.Error(codes.Uninitialized, err.Error())
+	}
 
 	// if SrcAnDstCheckEnabled is false then disable the SrcAndDestCheck on running NAT instance
 	if providerSpec.SrcAndDstChecksEnabled != nil && !*providerSpec.SrcAndDstChecksEnabled {
```
Comment (rishabh-11) [inline pkg/aws/core.go:275]:
```suggestion
	if providerSpec.SrcAndDstChecksEnabled != nil && !*providerSpec.SrcAndDstChecksEnabled && *requiredInstance.SourceDestCheck {
```
No need to call `disableSrcAndDestCheck` if it is already disabled.

**Review by rishabh-11 — CHANGES_REQUESTED** [review]
minor change needed. Rest looks good.

**Review by rishabh-11 — APPROVED** [review]
/lgtm

**Pull request merged**